aten/src/ATen/native/nested/NestedTensorUtils.h

```diff
@@ -32,11 +32,14 @@ struct NestedTensorImpl;
 // The following functions are used to construct nested tensors from buffers and
 // metadata.
 
-inline at::Tensor wrap_buffer(
-    at::Tensor buffer,
-    at::Tensor nested_sizes) {
-  TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
-      buffer.is_contiguous(), "Given buffer must be contiguous.");
+inline at::Tensor wrap_buffer(at::Tensor buffer, at::Tensor nested_sizes) {
```

Comment:
There is also this internal assert:
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/NestedTensorImpl.cpp#LL204C1-L209C2